metadata-ingestion/src/datahub/configuration/validate_field_removal.py

```diff
@@ -0,0 +1,22 @@
+import warnings
+from typing import Type
+
+import pydantic
+
+
+def pydantic_removed_field(
+    field: str,
+    print_warning: bool = True,
+) -> classmethod:
+    def _validate_field_rename(cls: Type, values: dict) -> dict:
+        if field in values:
+            if print_warning:
+                warnings.warn(
+                    f"The {field} was removed, please remove it from your recipe.",
+                    UserWarning,
```

Comment:
What is this UserWarning thing and where is it defined?

Reply:
UserWarning is a python builtin, similar to FutureWarning, DeprecationWarning, etc.

DeprecationWarning is only for developer facing stuff (e.g. calls a deprecated method) and doesn't always show up, so I tend to use UserWarning instead.